### PR TITLE
Update docker documentation

### LINKED ISSUE #1540: Docs page "Usage Guide > Run garage with Docker"
See #1426

### PR DESCRIPTION
**Pull request by gitanshu**
closes #1540

### CODE REVIEW

File: docs/user/docker.md [hunk 1]
```diff
@@ -30,91 +45,100 @@ To save the experiment data generated in the container, you need to specify a
...
+
 ```
-make build-headless ADD_ARGS="--build-arg MY_VAR=123"
-make run-headless ADD_ARGS="-e MY_VAR=123"
+make build-headless BUILD_ARGS="--build-arg MY_VAR=123"
+make run-headless RUN_ARGS="-e MY_VAR=123"
 ```
 
-#### Prerequisites
+## nvidia image
```
Comment (krzentner) [inline docs/user/docker.md:111]: It reads kinda weird that this subsection isn't capitalized.

Comment (gitanshu) [reply docs/user/docker.md]: Capitalized now

File: docs/user/docker.md [hunk 2]
```diff
@@ -30,91 +45,100 @@ To save the experiment data generated in the container, you need to specify a
...
+
 ```
-make build-headless ADD_ARGS="--build-arg MY_VAR=123"
-make run-headless ADD_ARGS="-e MY_VAR=123"
+make build-headless BUILD_ARGS="--build-arg MY_VAR=123"
+make run-headless RUN_ARGS="-e MY_VAR=123"
 ```
 
-#### Prerequisites
+## Nvidia image
```
Comment (ryanjulian) [inline docs/user/docker.md:111]: NVIDIA

File: docs/user/docker.md [hunk 3]
```diff
@@ -30,91 +45,100 @@ To save the experiment data generated in the container, you need to specify a
...
 
-### Build and run the image
+### Build and run the nvidia image
 
 The same rules for the headless image apply here, except that the target names
 are the following:
+
 ```
 make build-nvidia
 make run-nvidia
```
Comment (ryanjulian) [inline docs/user/docker.md:143]: doesn't run-nvidia automatically run `build-nvidia`?

Comment (gitanshu) [reply docs/user/docker.md:143]: It does. So should we not put both build-headless and build-nvidia in here?

Comment (ryanjulian) [reply docs/user/docker.md:143]: yeah -- so just let `make` do the work.

**Pull request merged**